### PR TITLE
Tests for worklog-export

### PR DESCRIPTION
**Pull request by stkofler**
# Description

createRows() from worklog-export has been extracted to a new file and initial tests have been implemented.

Note: I amended the suggested changes for start/endTimes initialization.

### CODE REVIEW

File: src/app/features/worklog/worklog-export/worklog-export.util.ts [hunk 1]
```diff
@@ -0,0 +1,274 @@
...
+  tasks: WorklogTask[];
+  titles?: string[];
+  titlesWithSub?: string[];
+  notes: string[];
+  projects: string[];
+  tags: string[];
+}
+
+export const createRows = (tasks: WorklogTask[], startTimes: WorkStartEnd, endTimes: WorkStartEnd,
+  groupBy: WorklogGrouping, allProjects: Project[], allTags: Tag[]): RowItem[] => {
```
Comment (johannesjo) [inline src/app/features/worklog/worklog-export/worklog-export.util.ts:31]: Probably makes sense to convert the params to a single object.

File: src/app/features/worklog/worklog-export/worklog-export.util.ts [hunk 2]
```diff
@@ -0,0 +1,274 @@
...
+
+  const mapToGroups = (task: WorklogTask) => {
+    const taskGroups: { [key: string]: RowItem } = {};
+    const createEmptyGroup = (): RowItem => {
```
Comment (johannesjo) [inline src/app/features/worklog/worklog-export/worklog-export.util.ts:35]: İ'd prefer moving this out of this function.

File: src/app/features/worklog/worklog-export/worklog-export.util.ts [hunk 3]
```diff
@@ -0,0 +1,274 @@
...
+            workEnd: endTimes[day],
+            timeSpent: 0,
+            timeEstimate: 0
+          };
+          if (!task.subTaskIds || task.subTaskIds.length === 0) {
+            taskGroups[day].timeSpent = task.timeSpentOnDay[day];
+            taskGroups[day].timeEstimate = task.timeEstimate / numDays;
+          }
+        });
+        break;
```
Comment (johannesjo) [inline src/app/features/worklog/worklog-export/worklog-export.util.ts:82]: Maybe using 
`return workGroups;`

Makes it easier to grasp what is happening.

File: src/app/features/worklog/worklog-export/worklog-export.util.ts [hunk 4]
```diff
@@ -0,0 +1,274 @@
...
+  titlesWithSub?: string[];
+  notes: string[];
+  projects: string[];
+  tags: string[];
+}
+
+export const createRows = (tasks: WorklogTask[], startTimes: WorkStartEnd, endTimes: WorkStartEnd,
+  groupBy: WorklogGrouping, allProjects: Project[], allTags: Tag[]): RowItem[] => {
+
+  const mapToGroups = (task: WorklogTask) => {
```
Comment (johannesjo) [inline src/app/features/worklog/worklog-export/worklog-export.util.ts:33]: İ would move this out of the parent function. İt's way too fat :)

**Comment by johannesjo** [comment]
Much better already! Thank you very much once more. Breaking down `createRows`  into smaller pieces might improve the code's readability even more.

**Comment by stkofler** [comment]
> Much better already! Thank you very much once more. Breaking down `createRows` into smaller pieces might improve the code's readability even more.

I totally agree on all of your points here. I'm actually already working on it as there are a few more refactorings like these which I would like to suggest soon. I just wanted to get the testing part approved so I know I'm on the right track. :)
Now it's up to you if you want to merge this first and I'll open a new PR with refactoring changes or if you want me to continue working on this here...

**Comment by johannesjo** [comment]
I am merging this then :)

**Pull request merged**